homeassistant/components/airios/config_flow.py

```diff
@@ -0,0 +1,629 @@
+                vol.Required(CONF_ADDRESS, default=conf_modbus_address): int,
+            }
+        )
+        return self.async_show_form(
+            step_id="network",
+            data_schema=schema,
+            errors=errors,
+        )
+
+    async def async_step_serial(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Step when setting up serial configuration."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            modbus_address = user_input[CONF_ADDRESS]
+            user_selection = user_input[CONF_DEVICE]
+            if user_selection == CONF_MANUAL_PATH:
+                self._conf_modbus_address = modbus_address
+                return await self.async_step_serial_manual_path()
+
+            dev_path = await self.hass.async_add_executor_job(
+                usb.get_serial_by_id, user_selection
+            )
+            try:
+                data = await self._async_validate_bridge_serial(
+                    device=dev_path, modbus_address=modbus_address
+                )
+            except UnexpectedProductId:
+                errors["base"] = "unexpected_product_id"
+            except AiriosException:
+                errors["base"] = "cannot_connect"
+            else:
+                return await self._finish(data)
+
+        ports = await self.hass.async_add_executor_job(serial.tools.list_ports.comports)
+        list_of_ports = {
+            port.device: usb.human_readable_device_name(
+                port.device,
+                port.serial_number,
+                port.manufacturer,
+                port.description,
+                f"{port.vid}" if port.vid else None,
+                f"{port.pid}" if port.pid else None,
+            )
+            for port in ports
+        }
+
+        list_of_ports[CONF_MANUAL_PATH] = CONF_MANUAL_PATH
+        conf_device = vol.UNDEFINED
+        conf_modbus_address = CONF_DEFAULT_SERIAL_MODBUS_ADDRESS
+        if self.source == SOURCE_RECONFIGURE:
+            if hasattr(self._reconfigure_data, CONF_DEVICE):
+                conf_device = self._reconfigure_data[CONF_DEVICE]
+            if hasattr(self._reconfigure_data, CONF_ADDRESS):
+                conf_modbus_address = self._reconfigure_data[CONF_ADDRESS]
+
+        schema = vol.Schema(
+            {
+                vol.Required(CONF_DEVICE, default=conf_device): vol.In(list_of_ports),
+                vol.Required(CONF_ADDRESS, default=conf_modbus_address): int,
+            }
+        )
+        return self.async_show_form(
+            step_id="serial",
+            data_schema=schema,
+            errors=errors,
+        )
+
+    async def async_step_serial_manual_path(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Select path manually."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            modbus_address = self._conf_modbus_address
```

Comment:
We generally name our constants CONF_*, so let's not name our variables with `conf`